runtime/v2/runc/v2/service.go

```diff
@@ -666,7 +666,8 @@ func (s *service) Shutdown(ctx context.Context, r *taskAPI.ShutdownRequest) (*pt
 	defer s.mu.Unlock()
 
 	// return out if the shim is still servicing containers
-	if len(s.containers) > 0 {
+	if !r.Now && len(s.containers) > 0 {
+		logrus.Debug("Not shutting down due to active containers")
```

Comment:
Sorry, bit random comment, but I keep looking at these `logrus` line, and let me at least leave a comment 😅 

Should these use `log.G(ctx)` ? Looking at this code, the context should have a logger attached: https://github.com/containerd/containerd/blob/0bcd45d3a0f4a09bdb4dd6010d2fada08aca472e/runtime/v2/shim.go#L270-L272

Same here; `handleSignals` is passed a new logger; https://github.com/containerd/containerd/blob/0bcd45d3a0f4a09bdb4dd6010d2fada08aca472e/runtime/v2/shim/shim.go#L213-L217

But the context also already has a logger attached; https://github.com/containerd/containerd/blob/0bcd45d3a0f4a09bdb4dd6010d2fada08aca472e/runtime/v2/shim/shim.go#L204

Reply:
+1 It would be better to take a logger from the context for consistency.

Reply:
The reason I did not do this is because nothing around it is doing this, but I definitely can.

Reply:
Yeah, everything in the v2 runc shim is using logrus directly.